src/services/stringCompletions.ts

```diff
@@ -261,9 +261,12 @@ namespace ts.Completions.StringCompletions {
 
         const uniques = new Map<string, true>();
         const candidates: Signature[] = [];
-        checker.getResolvedSignature(argumentInfo.invocation, candidates, argumentInfo.argumentCount);
+        const invocation = argumentInfo.invocation as NewExpression | CallExpression;
+        // omit trailing comma from the argument count
+        const argumentCount = argumentInfo.argumentCount - (invocation.arguments?.hasTrailingComma ? 1 : 0);

```

Comment:
There's something a little bit backwards here - we are leveraging a bunch of the logic of signature help, which I'm pretty sure gives us the wrong `argumentCount` because it has to account for signature help for when a user is in the last position of a signature and you intend to add an argument; but because of that, we have to then re-adjust the argument count. So we're just hacking on the hack of signature help.

I would love to separate completions logic out from `SignatureHelp` more, but I'd rather not see the full refactor in this PR. It seems like the adjustment should actually happen in `signatureHelp.ts` - specifically in `getArgumentInfoForCompletions` (https://github.com/microsoft/TypeScript/blob/1fe9bfdd0ee0ecf9dff23a40c0066645719e46b9/src/services/signatureHelp.ts#L166).

Reply:
@DanielRosenwasser Could you look at my previous comment?

Reply:
Well I think that `getArgumentInfoForCompletions` is the place that should be doing the decrementing - only string literal completions relies on it right?

Reply:
Also, `completions.ts` relies on it

Reply:
Added changes to `signatureHelp`. If the position is in the string "/**/", then the comma of the last child element is ignored. Does that make sense?

Reply:
@DanielRosenwasser I've renamed the parameter.